code/datums/components/_component.dm

```diff
@@ -534,3 +534,9 @@
   */
 /datum/component/ui_host()
 	return parent
+
+///Where atoms should drop if taken from this component
+/datum/component/drop_location()
+	var/atom/parent_atom = parent
+	return parent_atom.drop_location()

```

Comment:
couldn't the parent be a datum, though?

Reply:
This for things that connected to atoms, storages, mechas, etc. 
If something call itt to datum, something going wrong and we know this.

Reply:
i move this from /datum/component/construction/proc/drop_location()

Reply:
remove the `var/atom/parent_atom = parent` and
```suggestion
	return parent.drop_location()

```

Reply:
now it work, before it say that datum dont have drop_location()